float32 distance
---
float32 distance
